tools/apiview/parsers/CONTRIBUTING.md

```diff
@@ -0,0 +1,103 @@
+- `Children (c)` : The nodes immediate children. For a namespace this would be classes, for a class this would be the class constructors and methods.
+
+### StructuredToken
+- `Value (v)` : The token value which will be dispalyed. Can be null.  Spacing tokens don't need to have value.
+- `Id (i)` : This is essentially whatever existing parser was asigning to the token DefinitionId. You dont have to assign a tokenId.
+- `Kind (k)` : An enum
+  - `Content` Specifies that the token is content
+  - `LineBreak` Space token indicating switch to new line.
+  - `NoneBreakingSpace` Regular single space
+  - `TabSpace` 4 NoneBreakingSpaces
+  - `ParameterSeparator` Use this between method parameters. Depending on user setting this would result in a singlespace or new line
+  - `Url` A url token should have `LinkText` property and the url/link should be the token value.
+  All tokens should be content except for spacing tokens and url. ParameterSeparator should be used between method or function parameters. 
+- `Tags (t)` : Use this for opt in or opt out boolean properties.The currently supported tags are 
+  - `SkippDiff`  Indicate that a token should not be used in computatation of diff.
+  - `Deprecated` Mark a token as deprecated
+- `Properties (p)` : Properties of the token.
+  - `GroupId` : `doc` to group consecutive comment tokens as documentation.
+  - `NavigateToId` Id for navigating to where the node where the token is defined. Should match the definitionId of the node.
+- `RenderClasses (rc)` : Add css classes for how the tokens will be rendred. Classes currently being used are `text` `keyword` `punc` `tname` `mname` `literal` `sliteral` `comment` Feel free to add your own custom class. Whatever custom classes you use please provide us the appriopriate css for the class so we can update APIView.
+
+Json property names for each property is shown in brackets.
+
+Dont worry about indentation that will be handeled by the tree structure, unless you want to have indentation between the tokens then use `TabSpace` token kind.
+
+If your packages contains multiple assemblies then you will have multiple trees with multiple roots. Assign the final parsed value to a `List<APITreeNode> APIForest` property of the `CodeFile`.
+
+Serialize the generated code file to JSON. Try to make the json as small as possible by ignoring null values and empty collections, and using the abbreviated names of theproperties as the Json property name
+
+## How to handle commons Scenarios
+- TEXT, KEYWORD, COMMENT : Add `text`, `keyword`, `comment` to RenderClasses of the token
+- NEW_LINE : Create a token with `Kind = LineBreak`
+- WHITE_SPACE :  Create token with `Kind = NoneBreakingSpace`
+- PUNCTUATION : Create a token with `Kind = Content` and the `Value = the punctuation`
+- DOCUMENTATION : Add `GroupId = doc` in the properties of the token. This identifies a range of consecutive tokens as belonging to a group.
```

Comment:
Doesn't being in a tree node also do this? Also, why would I set `GroupId = doc`? Why not just have a tag called "Documentation" when can be set?

Reply:
Use `GroupId = doc` for contiguous comment tokens that are documentation. Potentially GroupId can be used to group range of other similar tokens.